pkg/reactable/bloom.go

```diff
@@ -78,29 +80,38 @@ type DendroCore struct {
 
 func (r *Reactable) addBloomGadget(a *combat.AttackEvent) {
 	r.core.Tasks.Add(func() {
-		var t combat.Gadget = NewDendroCore(r.core, r.self, a)
+		var t combat.Gadget = NewDendroCore(r.core, r.self.Shape(), a)
 		r.core.Combat.AddGadget(t)
 		r.core.Events.Emit(event.OnDendroCore, t, a)
 	}, DendroCoreDelay)
 }
 
-func NewDendroCore(c *core.Core, pos combat.Positional, a *combat.AttackEvent) *DendroCore {
+func NewDendroCore(c *core.Core, shp combat.Shape, a *combat.AttackEvent) *DendroCore {
 	s := &DendroCore{
 		srcFrame: c.F,
 	}
 
-	x, y := pos.Pos()
-	// for simplicity, seeds spawn randomly within 1 radius of target
-	x = x + 2*c.Rand.Float64() - 1
-	y = y + 2*c.Rand.Float64() - 1
-	s.Gadget = gadget.New(c, core.Coord{X: x, Y: y, R: 0.2}, combat.GadgetTypDendroCore)
+	circ, ok := shp.(*combat.Circle)
+	if !ok {
+		panic("rectangle target hurtbox is not supported for dendro core spawning")
+	}
+
+	// for simplicity, seeds spawn randomly at radius + 0.5
+	// https://stackoverflow.com/questions/9879258/how-can-i-generate-random-points-on-a-circles-circumference-in-javascript
+	angle := c.Rand.Float64() * 2 * math.Pi
+	r := circ.Radius()
+	pos := combat.Point{
+		X: math.Cos(angle) * (r + 0.5),
+		Y: math.Sin(angle) * (r + 0.5),
+	}
+	s.Gadget = gadget.New(c, pos, 0.2, combat.GadgetTypDendroCore)
```

Comment:
fine with current implementation. Thinking out loud wonder if it would be faster if you instead:
1. generate random point inside unit circle (using rejection sampling, maybe make a helper function in `point.go` since I imagine generating point within a radius will be common)
2. normalize that point
3. multiply unit vector by `r`

Reply:
This is supposed to generate a point specifically on the circumference of the circle and not just any point inside of the circle, so I don't think this will be any faster.

Reply:
The `normalize` call would convert the point from inside the circle -> on the circle circumference (since `normalize` will make the vector have a magnitude of 1). Then multiplying a unit vector by a scalar `r` will make a new vector with the same direction and `magnitude = r`